static/js/components/OrderSummary.js

```diff
@@ -36,20 +47,40 @@ class OrderSummary extends React.Component {
     return <p className="payment-explanation">{text}</p>;
   }
 
+  showAmount(description: string, amount: string): Array<React$Element<*>> {
+    return [
+      <Cell col={8} tablet={4} phone={2} className="description" key={description}>
+        {description}
+      </Cell>,
+      <Cell
+        col={2}
+        key={`${description}-amount`}
+        className="align-right">
+        <b className="amount">{amount}</b>
+      </Cell>
+    ];
+  }
+
   render() {
-    let { course, courseRun, checkout, checkoutStatus } = this.props;
+    let { course, courseRun, checkout, checkoutStatus, discount } = this.props;
+    let discountInfo;
+    if (discount){
```

Comment:
Can you change this to `discount !== undefined && discount !== null`, just to be explicit about the `0` case? We shouldn't have coupons that give $0 off but if we do it would be better to be explicit about it to the user